datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaProducerWrapper.java

```diff
@@ -177,7 +182,12 @@ void assignTask(DatastreamTask task) {
   }
 
   void unassignTask(DatastreamTask task) {
-    _tasks.remove(task);
+    boolean taskPresent = _tasks.remove(task);
+    // whenever a task is unassigned the kafka producer should be shutdown to ensure that
+    // there are no in-flight sends. Any further send will not work.
```

Comment:
nit: "Any further sends by this task will not work"

Also, do you think we should add a small sentence about how this affects existing tasks that share the same producer? i.e. their inflight records will attempt to flushed, but if it doesn't complete, they'll see send failures, but this is okay since they'll be able to create a new producer on the next send?